ldi-orchestrator/ldio-application/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/config/FlowAutoConfiguration.java

```diff
@@ -30,33 +31,25 @@ public FlowAutoConfiguration(ApplicationContext applicationContext) {
 	}
 
 	@Bean
-	@ConditionalOnMissingBean(LdiInput.class)
-	@DependsOn({ "componentExecutor" })
-	public LdiInput ldtoInput(OrchestratorConfig orchestratorConfig) {
-		return (LdiInput) getLdiComponent(orchestratorConfig.getInput().getName(),
-				orchestratorConfig.getInput().getConfig());
+	public List<Object> ldtoInput(OrchestratorConfig config) {
+		return config.getPipelines()
+				.stream()
+				.map(this::getLdiInput)
+				.collect(Collectors.toList());
 	}
 
-	@Bean("componentExecutor")
-	public ComponentExecutor componentExecutor(final OrchestratorConfig orchestratorConfig) {
-		List<LdiTransformer> ldiTransformers = orchestratorConfig.getTransformers()
+	public ComponentExecutor componentExecutor(final PipelineConfig pipelineConfig) {
+		List<LdiTransformer> ldiTransformers = pipelineConfig.getTransformers()
 				.stream()
 				.map(this::ldtoTransformer)
 				.toList();
-		List<LdiOutput> ldiOutputs = orchestratorConfig.getOutputs()
+		List<LdiOutput> ldiOutputs = pipelineConfig.getOutputs()
 				.stream()
 				.map(this::ldtoOutput)
 				.toList();
 		return new ComponentExecutorImpl(ldiTransformers, ldiOutputs);
 	}
 
-	@Bean
-	public LdiAdapter ldiAdapter(final OrchestratorConfig orchestratorConfig) {
-		ComponentDefinition adapterDefinition = orchestratorConfig.getInput().getAdapter();
-
-		return (LdiAdapter) getLdiComponent(adapterDefinition.getName(), adapterDefinition.getConfig());
-	}
-
 	private LdiTransformer ldtoTransformer(ComponentDefinition componentDefinition) {
```

Comment:
ldtoTransformer -> ldioTransformer

Reply:
same for ldtoOutput. Since these are not beans anymore perhaps make the naming more explicit of what the method does? 'get', 'create', 'configure' something like that?